packages/bootstrap-chat/src/composed/ChatApp.jsx

```diff
@@ -75,39 +78,55 @@ function ChatApp({
   }
 
   function trackAgentName(agentName) {
-    if (agentName) {
-      const previouslyTrackedNames = taskContext.currentAgentNames || {};
-      const newAgentName = { [agentId]: agentName };
-      const currentAgentNames = { ...previouslyTrackedNames, ...newAgentName };
-      updateContext({ currentAgentNames: currentAgentNames });
-    }
+    const previouslyTrackedNames = taskContext.currentAgentNames || {};
+    const newAgentName = { [agentId]: agentName, agentName: agentName };
+    const currentAgentNames = { ...previouslyTrackedNames, ...newAgentName };
+    updateContext({ currentAgentNames: currentAgentNames });
   }
 
   let mephistoProps = useMephistoLiveTask({
-    onStateUpdate: ({ state, status }) => {
-      trackAgentName(state.agent_display_name);
-      if (state.task_done) {
-        setInputMode(INPUT_MODE.DONE);
-      } else if (
+    onStatusUpdate: ({ status }) => {
+      if (
         [
           AGENT_STATUS.DISCONNECT,
           AGENT_STATUS.RETURNED,
           AGENT_STATUS.EXPIRED,
           AGENT_STATUS.TIMEOUT,
+          AGENT_STATUS.PARTNER_DISCONNECT,
           AGENT_STATUS.MEPHISTO_DISCONNECT,
         ].includes(status)
       ) {
         setInputMode(INPUT_MODE.INACTIVE);
-      } else if (state.wants_act) {
+        updateContext({
+          done_text: STATUS_TO_TEXT_MAP[status],
+          task_done: status == AGENT_STATUS.PARTNER_DISCONNECT,
```

Comment:
perhaps we can use this opportunity to clean up some of the names here: e.g. `isTaskDone` to denote boolean and `doneText` to keep consistent with the js camel casing? however if this variable is coming from Python, then it's probably fine to keep as is. I'm unsure if these new updates remove these from the backend entirely.

Reply:
Also this doesn't make much sense to me `task_done: status == AGENT_STATUS.PARTNER_DISCONNECT`

why is a task done when the partner disconnects? done in my head meant finished to completion, perhaps here we're signifying that the task failed and is now "done" in that sense? also what about the other error status cases other than partner disconnect?

Reply:
In this case, it's because all of the other disconnect statuses are "incomplete" in that the agent can't submit, but `PARTNER_DISCONNECT` allows submission b/c it wasn't this worker's fault.

I can clean up the casing for these though!